packages/flutter/test/cupertino/dialog_test.dart

```diff
@@ -103,116 +89,92 @@ void main() {
-            return new RaisedButton(
-              onPressed: () {
-                showDialog<void>(
-                  context: context,
-                  builder: (BuildContext context) {
-                    return new MediaQuery(
-                      data: MediaQuery.of(context).copyWith(textScaleFactor: 3.0),
-                      child: new CupertinoAlertDialog(
-                        title: const Text('The Title'),
-                        content: new Text('Very long content ' * 20),
-                        actions: const <Widget>[
-                          const CupertinoDialogAction(
-                            child: const Text('Cancel'),
-                          ),
-                          const CupertinoDialogAction(
-                            isDestructiveAction: true,
-                            child: const Text('OK'),
-                          ),
-                        ],
-                        scrollController: scrollController,
-                      ),
-                    );
-                  },
-                );
-              },
-              child: const Text('Go'),
-            );
-          }),
-        ),
-      )),
+      createAppWithButtonThatLaunchesDialog(
+        dialogBuilder: (BuildContext context) {
+          return new MediaQuery(
+            data: MediaQuery.of(context).copyWith(textScaleFactor: 3.0),
+            child: new CupertinoAlertDialog(
+              title: const Text('The Title'),
+              content: new Text('Very long content ' * 20),
+              actions: const <Widget>[
+                const CupertinoDialogAction(
+                  child: const Text('Cancel'),
+                ),
+                const CupertinoDialogAction(
+                  isDestructiveAction: true,
+                  child: const Text('OK'),
+                ),
+              ],
+              scrollController: scrollController,
+            ),
+          );
+        }
+      )
     );
 
     await tester.tap(find.text('Go'));
-
     await tester.pump();
-    await tester.pump(const Duration(seconds: 1));
 
     expect(scrollController.offset, 0.0);
     scrollController.jumpTo(100.0);
     expect(scrollController.offset, 100.0);
     // Set the scroll position back to zero.
     scrollController.jumpTo(0.0);
 
-    // Find the actual dialog box. The first decorated box is the popup barrier.
-    expect(tester.getSize(find.byType(DecoratedBox).at(1)), equals(const Size(270.0, 560.0)));
+    // Expect the modal dialog box to take all available height.
+    expect(
+      tester.getSize(
+        find.byKey(const Key('cupertino_alert_dialog_modal'))
+      ),
+      equals(const Size(270.0, 560.0)),
+    );
 
-    // Check sizes/locations of the text.
+    // Check sizes/locations of the text. The text is large so these 2 buttons are stacked.
+    // Visually the "Cancel" button and "OK" button are the same height. I don't know why
```

Comment:
You can debugDumpRenderTree in your test.

I suspect the GestureDetector are ultimately sized by their child which are different.

Reply:
I output the render tree.  I don't think the issue is gesture detector sizing.

The "OK" text reports a height of 50.0.  The "Cancel" text reports a height of 100.0.  That's the initial basis for the difference.  I have no idea why they're reporting different heights...

Check it out:

```
RenderParagraph#f6884 relayoutBoundary=up27 NEEDS-PAINT
             ╎                               │                                         │ creator: RichText ← Text ← DefaultTextStyle ← Align ← Padding ←
             ╎                               │                                         │   Container ← ConstrainedBox ← Listener ← _GestureSemantics ←
             ╎                               │                                         │   RawGestureDetector ← GestureDetector ← CupertinoDialogAction ←
             ╎                               │                                         │   ⋯
             ╎                               │                                         │ parentData: offset=Offset(60.5, 0.0) (can use size)
             ╎                               │                                         │ constraints: BoxConstraints(0.0<=w<=222.0, 0.0<=h<=Infinity)
             ╎                               │                                         │ size: Size(101.0, 50.0)
             ╎                               │                                         │ textAlign: center
             ╎                               │                                         │ textDirection: ltr
             ╎                               │                                         │ softWrap: wrapping at box width
             ╎                               │                                         │ overflow: clip
             ╎                               │                                         │ textScaleFactor: 3.0
             ╎                               │                                         │ locale: en_US
             ╎                               │                                         │ maxLines: unlimited
             ╎                               │                                         ╘═╦══ text ═══
             ╎                               │                                           ║ TextSpan:
             ╎                               │                                           ║   inherit: false
             ╎                               │                                           ║   color: Color(0x80ff3b30)
             ╎                               │                                           ║   family: .SF UI Text
             ╎                               │                                           ║   size: 16.8
             ╎                               │                                           ║   weight: 400
             ╎                               │                                           ║   baseline: alphabetic
             ╎                               │                                           ║   "OK"
             ╎                               │                                           ╚═══════════


RenderParagraph#20d39 relayoutBoundary=up27 NEEDS-PAINT
             ╎                               │                         │               │ creator: RichText ← Text ← DefaultTextStyle ← Align ← Padding ←
             ╎                               │                         │               │   Container ← ConstrainedBox ← Listener ← _GestureSemantics ←
             ╎                               │                         │               │   RawGestureDetector ← GestureDetector ← CupertinoDialogAction ←
             ╎                               │                         │               │   ⋯
             ╎                               │                         │               │ parentData: offset=Offset(0.0, 0.0) (can use size)
             ╎                               │                         │               │ constraints: BoxConstraints(0.0<=w<=222.0, 0.0<=h<=Infinity)
             ╎                               │                         │               │ size: Size(222.0, 100.0)
             ╎                               │                         │               │ textAlign: center
             ╎                               │                         │               │ textDirection: ltr
             ╎                               │                         │               │ softWrap: wrapping at box width
             ╎                               │                         │               │ overflow: clip
             ╎                               │                         │               │ textScaleFactor: 3.0
             ╎                               │                         │               │ locale: en_US
             ╎                               │                         │               │ maxLines: unlimited
             ╎                               │                         │               ╘═╦══ text ═══
             ╎                               │                         │                 ║ TextSpan:
             ╎                               │                         │                 ║   inherit: false
             ╎                               │                         │                 ║   color: Color(0x80007aff)
             ╎                               │                         │                 ║   family: .SF UI Text
             ╎                               │                         │                 ║   size: 16.8
             ╎                               │                         │                 ║   weight: 400
             ╎                               │                         │                 ║   baseline: alphabetic
             ╎                               │                         │                 ║   "Cancel"
             ╎                               │                         │                 ╚═══════════
```

It's as if the render logic thinks that "Cancel" is wrapped, but to a 2nd line, but it isn't.

Reply:
Try swapping the names of the buttons, or changing the text.  Perhaps it is a problem in the text wrapping code, and it's adding just a newline because "Cancel" is just the right size.  Or maybe turn off wrapping on that text.